descqa/NumberDensityVersusRedshift.py

```diff
@@ -0,0 +1,307 @@
+from __future__ import print_function, unicode_literals, absolute_import 
+import os
+import numpy as np
+import re
+import math
+from GCR import GCRQuery
+from .base import BaseValidationTest, TestResult
+from .plotting import plt
+
+__all__ = ['NumberDensityVersusRedshift']
+
+#observations
+Coil2004 = 'Coil2004'
+Coil2004_magbin = 'Coil2004_magbin'
+Coil2004_maglim = 'Coil2004_maglim'
+DEEP2_JAN = 'DEEP2_JAN'
+Coil2004_lbl = 'Coil et. al. 2004'
+DEEP2_JAN_lbl = 'DEEP2 (JAN, p.c.)'
+
+#setup dict with parameters needed to read in validation data 
+data_info = {
+    'Coil2004_magbin': {'filename_template': 'N_z/DEEP2/Coil_et_al_2004_Table3_{}.txt', 'usecols':[0,1,2,4], 'skiprows':2},
+    'Coil2004_maglim': {'filename_template': 'N_z/DEEP2/Coil_et_al_2004_Table4_{}.txt', 'usecols':[0,1,2], 'skiprows':3},
+    'DEEP2_JAN':{'filename_template': 'N_z/DEEP2/JANewman_{}.txt','usecols':[0,1,2,3], 'skiprows':1},
+}
+
+#plotting constants
+figx_p = 9
+figy_p = 11
+lw2 = 2
+fsize = 16
+lsize = 10  
+default_colors = ['blue','r','m','g','navy','y','purple','gray','c','orange','violet','coral','gold','orchid','maroon','tomato','sienna','chartreuse','firebrick','SteelBlue']
+validation_color = 'black'
+
+class NumberDensityVersusRedshift(BaseValidationTest):
+    """
+    validation test to show N(z) distributions
+    """
+    def __init__(self, z='redshift_true', band='i', N_zbins=44,zlo=0., zhi=1.1, observation='', m_lo=27, m_hi=18, ncolumns=2, normed=True, **kwargs):
```

Comment:
one space after comma (before `zlo`)

Remove `m_lo` and `m_hi`

Reply:
Keeping m_lo and m_hi for case with no observations